controller/appcontroller.go

```diff
@@ -96,9 +97,66 @@ func (a CompareWith) Pointer() *CompareWith {
 	return &a
 }
 
+type CrdMetadata struct {
+	apiVersion string
+	kind       string
+	namespace  string
```

Comment:
Is it better to wrap it in metadata?

```
Metadata   struct {
		Namespace string `json:"namespace"`
		Name      string `json:"name"`
	} `json:"metadata"`
```

Reply:
I explicitly used the current structure to have most minimal allocation possible. As only a few fields are used, which are accessed only via functions, I don't see the need for nested structs.